app/models/agent.rb

```diff
@@ -149,18 +149,23 @@ def update_event_expirations!
     end
   end
 
-  def trigger_web_request(params, method, format)
+  def trigger_web_request(params, method, format, headers)
     if respond_to?(:receive_webhook)
       Rails.logger.warn "DEPRECATED: The .receive_webhook method is deprecated, please switch your Agent to use .receive_web_request."
       receive_webhook(params).tap do
         self.last_web_request_at = Time.now
         save!
       end
-    else
+    elsif method(:receive_web_request).arity == 3
```

Comment:
It's a nice thought to avoid breaking custom agents, but as of now we do not support agents in gems and normally just ensure that we do not break agent configurations with updates. I think it would be fine to modify the existing agents and always assume an arity of 4. What do you think @cantino ?